backend/store/postgres.go

```diff
@@ -24,29 +36,109 @@ func NewPostgresStore() (*PostgresStore, error) {
+	return gorm.Open(postgres.Open(connStr), &gorm.Config{})
+}
+
+func NewUserStore() (UserStore, error) {
+	db, err := NewPostgresStore()
+	if err != nil {
+		return nil, err
+	}
+	err = db.AutoMigrate(&model.User{})
+	if err != nil {
+		return nil, err
+	}
+	return &UserStoreImpl{PostgresStore: PostgresStore{db: db}}, nil
+}
+
+func NewCardDeckStore() (CardDeckStore, error) {
+	db, err := NewPostgresStore()
 	if err != nil {
 		return nil, err
 	}
+	err = db.AutoMigrate(&model.Card{}, &model.Deck{})
+	if err != nil {
+		return nil, err
+	}
+	return &CardDeckStoreImpl{PostgresStore: PostgresStore{db: db}}, nil
+}
 
-	err = db.AutoMigrate(&model.Card{}, &model.Deck{}, &model.Group{}, &model.User{}, &model.GroupUser{})
+func NewCollaborationStore() (CollaborationStore, error) {
+	db, err := NewPostgresStore()
 	if err != nil {
 		return nil, err
 	}
+	err = db.SetupJoinTable(&model.User{}, "Groups", &model.GroupUserRole{})
+	if err != nil {
+		return nil, err
+	}
+	err = db.SetupJoinTable(&model.Group{}, "Users", &model.GroupUserRole{})
+	if err != nil {
+		return nil, err
+	}
+	err = db.AutoMigrate(&model.Group{})
+	if err != nil {
+		return nil, err
+	}
+	return &CollaborationStoreImpl{PostgresStore: PostgresStore{db: db}}, nil
+}
+
+func (s *UserStoreImpl) FindUserByEmail(email string) (user *model.User, err error) {
+	err = s.db.Where(model.User{Email: email}).First(&user).Error
+	return
+}
+
+func (s *UserStoreImpl) RegisterNewUser(newUser *model.User) error {
+	rsp := s.db.Create(&newUser)
+	return rsp.Error
+}
+
+func (s *CardDeckStoreImpl) CreateDeck(newDeck *model.Deck) error {
+	rsp := s.db.Create(&newDeck)
+	return rsp.Error
+}
+
+func (s *CardDeckStoreImpl) FindDeckByPublicID(publicID string) (deck *model.Deck, err error) {
+	err = s.db.Where(model.Deck{PublicID: publicID}).Preload("Cards").First(&deck).Error
+	return
+}
+
+func (s *CardDeckStoreImpl) CreateCard(newCard *model.Card) error {
+	rsp := s.db.Create(&newCard)
+	return rsp.Error
+}
+
+func (s *CardDeckStoreImpl) FindDecksByGroupID(groupID uint) (decks []model.Deck, err error) {
+	err = s.db.Where(model.GroupUserRole{GroupID: groupID}).Find(&decks).Error
+	return
+}
+
+func (s *CollaborationStoreImpl) CreateNewGroupWithAdmin(adminUserID uint, newGroup *model.Group) error {
+	rsp := s.db.Create(&newGroup)
+	if rsp.Error != nil {
+		return rsp.Error
+	}
+	rsp = s.db.Create(model.GroupUserRole{GroupID: newGroup.ID, UserID: adminUserID, RoleType: model.RoleAdmin})
+	return rsp.Error
+}
 
-	return &PostgresStore{db: db}, nil
+func (s *PostgresStore) FindGroupByPublicID(publicID string) (group *model.Group, err error) {
+	err = s.db.Where(model.Group{PublicID: publicID}).First(&group).Error
+	return
 }
 
-func (s *PostgresStore) FindUserByEmail(email string) (*model.User, error) {
-	var usr model.User
-	err := s.db.Where("email = ?", email).First(&usr).Error
+func (s *CollaborationStoreImpl) GetGroupUserRole(userID uint, groupID uint) (*model.RoleType, error) {
```

Comment:
same as above, it's most of the time a better idea to return a default-value for primitive types, than `nil`
```suggestion
func (s *CollaborationStoreImpl) GetGroupUserRole(userID uint, groupID uint) (model.RoleType, error) {
```

Reply:
Fixed in 1961c77d5ebb2872d0b10ec2c359bf0b8eb7f3dd